### ISSUE TITLE
Crash when PneumaticCraft Smart Chest attempts to push into Blood Altar

### ISSUE THREAD

**Issue by MuteTiefling** [WayofTime/BloodMagic#1846]
#### Issue Description:
Out of Bounds crash. Logs below.

#### What happens:
Crash when PneumaticCraft Smart Chest attempts to push into Blood Altar

#### What you expected to happen:
Nothing, there's already an item in the altar, so it shouldn't do anything. 

#### Steps to reproduce:

1. Place Stone in Blood Altar
2. Place Smart Chest next to altar
3. Place items in the Chest
4. Set the Chest to push into the altar (this happens naturally as you're attempting to toggle to pull mode)

____
#### Affected Versions (Do *not* use "latest"):

- BloodMagic: BloodMagic-1.16.4-3.1.2-17
- Minecraft: 1.16.5
- Forge: 36.2.2
- PneumaticCraft: pneumaticcraft-repressurized-1.16.5-2.14.1-245

Logs: 
[latest.log](https://github.com/WayofTime/BloodMagic/files/7130863/latest.log)

Not really sure if this is on BM or PNC side. Reporting to both.

**Comment by MuteTiefling** [comment]
https://github.com/TeamPneumatic/pnc-repressurized/issues/892

**Comment by WayofTime** [comment]
From a cursory examination, it seems that PneumaticCraft is trying to pull from slot 7 of the Blood Altar; which is concerning, since there is only one slot for the Altar. I'd assume it's a PneumaticCraft issue, but it seems like that'd be a weird check to miss...

**Comment by desht** [comment]
Yep, it's most likely a logic error on the PNC side.  My theory is that I've cached a slot index which then went invalid (e.g. used to be a chest, but it got replaced with a Blood Altar, or something like that). The Smart Chest keeps track of the last slot of the neighbour item handler it interacted with, to reduce inventory scanning.

You're probably safe to close this, I don't think BM is doing anything wrong here.
